graphql/query_resolver.go

```diff
@@ -167,7 +169,20 @@ func (r *queryResolver) Distros(ctx context.Context, onlySpawnable bool) ([]*res
 		}
 		distros = d
 	}
+
+	userHasDistroCreatePermissions := usr.HasPermission(gimlet.PermissionOpts{
```

Comment:
Changes look good -- I might recommend moving the bulk of this logic into a function in the distro collection so that you can write a unit test for the "not admin only" case but i do recognize this isn't a big change